backend/src/services/model.ts

```diff
@@ -376,3 +379,66 @@ export async function createModelCardFromTemplate(
   const revision = await _setModelCard(user, modelId, template.card.schemaId, 1, template.card.metadata)
   return revision
 }
+
+export async function saveImportedModelCard(modelCard: ModelCardRevisionInterface, sourceModelId: string) {
+  const model = await Model.findOne({
+    id: modelCard.modelId,
+  })
+  if (!model) {
+    throw NotFound(`The mirrored model ID found in the notification cannot be found.`, { modelId: modelCard.modelId })
+  }
+  if (!model.settings.mirror.sourceModelId) {
+    throw InternalError('Cannot import model card to non mirrored model.')
+  }
+  if (model.settings.mirror.sourceModelId !== sourceModelId) {
+    throw InternalError('The source model ID of the mirrored model does not match the model Id of the imported model', {
+      sourceModelId: model.settings.mirror.sourceModelId,
+      importedModelId: sourceModelId,
+    })
+  }
+
+  const schema = await findSchemaById(modelCard.schemaId)
+  try {
+    new Validator().validate(modelCard.metadata, schema.jsonSchema, { throwAll: true, required: true })
+  } catch (error) {
+    if (isValidatorResultError(error)) {
+      throw BadReq('Model metadata could not be validated against the schema.', {
+        schemaId: modelCard.schemaId,
+        validationErrors: error.errors,
+      })
+    }
+    throw error
+  }
+
+  return await ModelCardRevisionModel.findOneAndUpdate(
+    { modelId: modelCard.modelId, version: modelCard.version },
+    modelCard,
+    {
+      upsert: true,
+    },
+  )
+}
+
+export async function setLatestImportedModelCard(modelId: string) {
+  // Add check for mirrored model
+  const latestModelCard = await ModelCardRevisionModel.findOne({ modelId }).sort({ version: -1 }).limit(1)
```

Comment:
I don't think the `sort` and `limit` on this `findOne` is doing what you expect. `findOne` will return the first result that meets the condition, so you'll just be sorting a single result. I think if you replace `findOne` with `find` it'll work as expected.

Also consider adding in some error handling in the event we don't find anything.

Reply:
Updated but this method needs unit and manual testing

Reply:
Stuck with using `findOne` as this returns a single document rather than `find` as that returns a cursor. I have though removed the limit as that's redundant when using `findOne`